jobs/io/stub_io.py

```diff
@@ -19,52 +19,289 @@
 from __future__ import absolute_import  # pylint: disable=import-only-modules
 from __future__ import unicode_literals  # pylint: disable=import-only-modules
 
+import contextlib
+import pickle
+import threading
+import xmlrpc.client
+
+from jobs import job_utils
 import python_utils
 
 import apache_beam as beam
 
 
-class ModelIoStub(python_utils.OBJECT):
-    """Stubs datastore operations by reading from/writing to a simple dict."""
+class DatastoreioStub(python_utils.OBJECT):
+    """Stubs the apache_beam.io.gcp.datastore.v1new.datastoreio module.
+
+    The PTransforms returned by this stub use an XML-RPC server to operate on
```

Comment:
Maybe you could talk a bit more about what exactly the XML-RPC server does, etc., to help the reader develop a conceptual mental model of the role it plays?

Reply:
Done.